runhouse/resources/hardware/cluster.py

```diff
@@ -1946,3 +1950,162 @@ def _folder_mv(
 
     def _folder_exists(self, path: Union[str, Path]):
         return self.client.folder_exists(path=path)
+
+    @staticmethod
+    def get_clusters_from_den(cluster_filters: dict):
+        get_clusters_params = {
+            "resource_type": "cluster",
+            "folder": rns_client.username,
+        }
+
+        # send den request with filters if the user specifies filters.
+        # If "all" filter is specified - get all clusters (no filters are added to get_clusters_params)
+        if cluster_filters and "all" not in cluster_filters.keys():
+            get_clusters_params.update(cluster_filters)
+
+        # If not filters are specified, get only running clusters.
+        elif not cluster_filters:
+            get_clusters_params.update(
+                {"cluster_status": "running", "since": LAST_ACTIVE_AT_TIMEFRAME}
+            )
+
+        clusters_in_den_resp = rns_client.session.get(
+            f"{rns_client.api_server_url}/resource",
+            params=get_clusters_params,
+            headers=rns_client.request_headers(),
+        )
+
+        return clusters_in_den_resp
+
+    @staticmethod
+    def _get_unsaved_live_clusters(den_clusters: list[Dict], sky_live_clusters: list):
```

Comment:
similar note as above, these aren't really methods that are speciifc to a specific runhouse cluster object, probably make sense for them to live elsewhere

Reply:
yeah, makes sense. Moved to hardware utils file.